source/slang/slang-ir-hlsl-legalize.cpp

```diff
@@ -0,0 +1,94 @@
+// slang-ir-hlsl-legalize.cpp
+#include "slang-ir-hlsl-legalize.h"
+
+#include <functional>
+
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+#include "slang-ir-inst-pass-base.h"
+#include "slang-ir-specialize-function-call.h"
+#include "slang-ir-util.h"
+
+namespace Slang
+{
+
+void searchChildrenForForceVarIntoStructTemporarily(IRModule* module, IRInst* inst)
+{
+    for(auto child : inst->getChildren())
+    {
+        switch(child->getOp())
+        {
+        case kIROp_Block:
+        {
+            searchChildrenForForceVarIntoStructTemporarily(module, child);
+            break;
+        }
+        case kIROp_Call:
+        {
+            auto call = as<IRCall>(child);
+            for(UInt i = 0; i < call->getArgCount(); i++)
+            {
+                auto arg = call->getArg(i);
+                if(arg->getOp() != kIROp_ForceVarIntoStructTemporarily)
+                    continue;
+                auto forceStructArg = arg->getOperand(0);
+                auto forceStructBaseType = as<IRType>(forceStructArg->getDataType()->getOperand(0));
+                if(forceStructBaseType->getOp() == kIROp_StructType)
+                {
+                    arg->replaceUsesWith(arg->getOperand(0));
```

Comment:
should be `call->setOperand(i, arg->getOperand(0)` to avoid unexpected changes in other places.